### PR TITLE
Implement warning system for long parse times

### PR DESCRIPTION
**Pull request by APickledWalrus**
### Description
This PR implements a tiny warning that will occur if a Statement takes too long to parse.
By default, this warning is disabled. I've added a new config option for users to set the threshold if they wish to enable it.

---
**Target Minecraft Versions:** Any
**Requirements:** N/A
**Related Issues:**
- https://github.com/SkriptLang/Skript/issues/4759

### CODE REVIEW

File: src/main/java/ch/njol/skript/ScriptLoader.java [hunk 1]
```diff
@@ -1111,9 +1111,16 @@ public static ArrayList<TriggerItem> loadItems(SectionNode node) {
...
 					continue;
 
+				long start = System.currentTimeMillis();
 				Statement stmt = Statement.parse(expr, "Can't understand this condition/effect: " + expr);
 				if (stmt == null)
 					continue;
+				long requiredTime = SkriptConfig.longParseTimeWarningThreshold.value().getMilliSeconds();
+				if (requiredTime > 0 && System.currentTimeMillis() - start > requiredTime)
+					Skript.warning(
+						"The line '" + expr + "' took took a long time to parse."
```
Comment (TPGamesNL) [inline src/main/java/ch/njol/skript/ScriptLoader.java:1121]: Took took :)

Comment (APickledWalrus) [reply src/main/java/ch/njol/skript/ScriptLoader.java:1121]: Oh, that's what you mean by that! 😆

**Comment by TheLimeGlass** [comment]
What's the point of putting the default as 0 seconds? It'll never be used.

**Comment by TPGamesNL** [comment]
Because most users don't care if their script takes a bit more time to load. This is only for more advanced users who want to figure out which lines cause their script to take a long time to load.

@APickledWalrus Is the `expr` needed in the warning message? Isn't it already included because of the currentNode info

**Review by TheLimeGlass — APPROVED** [review]

**Comment by TheLimeGlass** [comment]
Another related issue https://github.com/SkriptLang/Skript/issues/1749

**Comment by TPGamesNL** [comment]
This PR doesn't solve that issue. The issue is about preventing runtime crashes (server hanging), e.g. with infinite loops. This PR is about parse-time, plus this PR won't prevent a crash.

**Comment by TheLimeGlass** [comment]
> This PR doesn't solve that issue. The issue is about preventing runtime crashes (server hanging), e.g. with infinite loops. This PR is about parse-time, plus this PR won't prevent a crash.

There is an infinitely repeating catch for runtime, so that makes no sense why that issue is open.

**Comment by TPGamesNL** [comment]
> There is an infinitely repeating catch for runtime, so that makes no sense why that issue is open.

If you're talking about the  'script infinitely repeated itself' message, that's only for `StackOverflowError`s [[1]](https://github.com/SkriptLang/Skript/blob/master/src/main/java/ch/njol/skript/lang/TriggerItem.java#L95), e.g. infinite recursive function calls, not infinite while loops.

**Comment by APickledWalrus** [comment]
> @APickledWalrus Is the `expr` needed in the warning message? Isn't it already included because of the currentNode info

Yep, it was included. I've updated the message to account for this.

**Review by TPGamesNL — APPROVED** [review]

**Pull request merged**